src/main/java/technology/dice/dicefairlink/iterators/RandomisedCyclicIterator.java

```diff
@@ -34,6 +35,17 @@ public boolean hasNext() {
     return !elements.isEmpty();
   }
 
+  public boolean hasSameContent(final Collection<T> externalElements) {
```

Comment:
On the constructor , you can do: `this.elements = new ArrayList<>(collection);`. Also there we should be consistent with the use of `this.` - either everywhere or nowhere

Reply:
Fixed.

Reply:
Fixed.

Reply:
Fixed.